src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs

```diff
@@ -147,13 +150,16 @@ private async Task UpdateRecentsList(NotifyCollectionChangedEventArgs e)
 		/// Add the RecentItem to the ObservableCollection for the UI to render.
 		/// </summary>
 		/// <param name="recentItem">The recent item to be added</param>
-		private async Task AddItemToRecentListAsync(RecentItem recentItem, int index = -1)
+		private bool AddItemToRecentList(RecentItem recentItem, int index = -1)
 		{
 			if (!recentItemsCollection.Any(x => x.Equals(recentItem)))
 			{
 				recentItemsCollection.Insert(index < 0 ? recentItemsCollection.Count : Math.Min(index, recentItemsCollection.Count), recentItem);
-				await recentItem.LoadRecentItemIcon();
+				_ = recentItem.LoadRecentItemIcon()
```

Comment:
Optional change, can revert if asked: do not wait for recent item icon to be loaded.

Reply:
Looks good